week-4/implement/count.test.js

```diff
@@ -1,5 +1,24 @@
 // implement a function countChar that counts the number of times a character occurs in a string
 
+function countChar(str, char) {
+  let index = 0;
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === char) {
```

Comment:
This is a well written code however can you explain this a bit? It seems a bit confusing to me, maybe if I understand what you have done here, it would make sense to me.